document/core/exec/instructions.rst

```diff
@@ -2635,14 +2635,23 @@ Control Instructions
 
 5. Pop the values :math:`\val^m` from the stack.
 
-6. Let :math:`H` be the :ref:`catching exception handler <syntax-handler>` :math:`\CATCHadm\{a_x~instr_2^\ast\}^\ast\{\epsilon~\instr_3^\ast\}^?`, whose mapping from tag addresses to catch bodies corresponds to the |CATCH| and |CATCHALL| clauses, where :math:`a_x` is the tag address of each tag :math:`x`.
+6. Let :math:`F` be the :ref:`current <exec-notation-textual>` :ref:`frame <syntax-frame>`.
 
-   .. todo::
-       Step 6 needs proper writing out into a loop over x*, which looks up each corresponding a_x in the store.
+7. Let :math:`H` be the empty :ref:`catching exception handler <syntax-handler>` :math:`\CATCHadm`.
 
-7. :ref:`Enter <exec-instr-seq-enter>` the block :math:`H~(\val^m~\instr_1^\ast)~\END` with label :math:`L`.
+8. For each catch clause :math:`(\CATCH~x~\instr_2^\ast)` do:
 
-8. :ref:`Enter <exec-handler-enter>` the block :math:`\val^m~\instr_1^\ast` with exception handler :math:`H`.
+   a. Let :math:`a_x` be the tag address :math:`F.\AMODULE.\MITAGS[x]`.
+
+   b. Append :math:`\{a_x~\instr_2^\ast\}` to :math:`H`.
+
+9. If there is a catch all clause :math:`(\CATCHALL~\instr_3^\ast)`, then
+
+   a. Append :math:`\{\epsilon~\instr_3^\ast\}` to :math:`H`.
+
+10. :ref:`Enter <exec-instr-seq-enter>` the block :math:`H~(\val^m~\instr_1^\ast)~\END` with label :math:`L`.
```

Comment:
This is a bit inconsistent with how block entering is described elsewhere (without the delimiters). The idea is that the prose does not actually write out the administrative delimiters.

Reply:
I think I understand now, what you mean with "header of a block-like construct". Ok, I will try to come up with a different formulation, although still I think the "parts" of a ` \CATCHadm` handler should not be also called handlers, but handler clauses or something similar. WDYT  @rossberg ?